src/test/e2e/50_oci_package_test.go

```diff
@@ -0,0 +1,158 @@
+
+	_, stdErr, err := e2e.execZarfCommand("tools", "registry", "login", "--username", suite.ZarfState.RegistryInfo.PushUsername, "-p", suite.ZarfState.RegistryInfo.PushPassword, suite.Reference.Registry)
+	require.NoError(t, err)
+	require.Contains(t, stdErr, "logged in", "failed to login to the registry")
+}
+
+func (suite *RegistryClientTestSuite) TearDownSuite() {
+	t := suite.T()
+	defer e2e.teardown(t)
+
+	suite.RegistryTunnel.Close()
+
+	stdOut, stdErr, err := e2e.execZarfCommand("package", "remove", "helm-oci-chart", "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+}
+
+func (suite *RegistryClientTestSuite) Test_0_Publish() {
+	t := suite.T()
+	t.Log("E2E: Package Publish oci://")
+
+	// Publish package.
+	example := filepath.Join(suite.PackagesDir, fmt.Sprintf("zarf-package-helm-oci-chart-%s-0.0.1.tar.zst", e2e.arch))
+	ref := suite.Reference.String()
+	stdOut, stdErr, err := e2e.execZarfCommand("package", "publish", example, "oci://"+ref, "--insecure")
+	require.NoError(t, err, stdOut, stdErr)
+	require.Contains(t, stdErr, "Published "+ref)
+
+	// Publish w/ package missing `metadata.version` field.
+	example = filepath.Join(suite.PackagesDir, fmt.Sprintf("zarf-package-dos-games-%s.tar.zst", e2e.arch))
+	_, stdErr, err = e2e.execZarfCommand("package", "publish", example, "oci://"+ref, "--insecure")
+	require.Error(t, err, stdErr)
+}
+
+func (suite *RegistryClientTestSuite) Test_1_Pull() {
+	t := suite.T()
+	t.Log("E2E: Package Pull oci://")
+
+	out := fmt.Sprintf("zarf-package-helm-oci-chart-%s-0.0.1.tar.zst", e2e.arch)
+
+	// Build the fully qualified reference.
+	suite.Reference.Repository = "helm-oci-chart"
+	suite.Reference.Reference = fmt.Sprintf("0.0.1-%s", e2e.arch)
+	ref := suite.Reference.String()
+
+	// Pull the package via OCI.
+	stdOut, stdErr, err := e2e.execZarfCommand("package", "pull", "oci://"+ref, "--insecure")
+	require.NoError(t, err, stdOut, stdErr)
+	require.Contains(t, stdErr, "Pulled "+ref)
+	defer e2e.cleanFiles(out)
+
+	// Verify the package was pulled.
+	require.FileExists(t, out)
+
+	// Test pull w/ bad ref.
+	stdOut, stdErr, err = e2e.execZarfCommand("package", "pull", "oci://"+badRef.String(), "--insecure")
+	require.Error(t, err, stdOut, stdErr)
+}
+
+func (suite *RegistryClientTestSuite) Test_2_Deploy() {
+	t := suite.T()
+	t.Log("E2E: Package Deploy oci://")
+
+	// Build the fully qualified reference.
+	suite.Reference.Repository = "helm-oci-chart"
+	suite.Reference.Reference = fmt.Sprintf("0.0.1-%s", e2e.arch)
+	ref := suite.Reference.String()
+
+	// Deploy the package via OCI.
+	stdOut, stdErr, err := e2e.execZarfCommand("package", "deploy", "oci://"+ref, "--insecure", "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+	require.Contains(t, stdErr, "Pulled "+ref)
+
+	stdOut, stdErr, err = e2e.execZarfCommand("tools", "kubectl", "get", "pods", "-n=helm-oci-demo", "--no-headers")
+	require.NoError(t, err, stdErr)
+	require.Contains(t, string(stdOut), "cool-name-podinfo")
```

Comment:
Just a note I don't think this is the release name for this example - this should just be the name from normal podinfo - see the oci helm chart fix for a way to test this.